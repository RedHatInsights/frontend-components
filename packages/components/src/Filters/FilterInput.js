import React, { Component } from 'react';
import { Checkbox, Radio } from '@patternfly/react-core';
import PropTypes from 'prop-types';

class FilterInput extends Component {
    state = {
        checked: false
    };

    componentDidMount() {
        const { type, filters, param, value } = this.props;
        switch (type) {
            case 'checkbox':
                param in filters && filters[param] && filters[param].includes(value)
                    ? this.setState({ checked: true }) : this.setState({ checked: false });
                break;
        }
    }

    handleChange = checked => {
        const { type, addRemoveFilters, param, value } = this.props;
        let updateState;
        switch (type) {
            case 'checkbox':
                updateState = { checked: !this.state.checked };
                break;
        }

        this.setState({ ...updateState });
        addRemoveFilters(value, param, type, checked);
    };

    render() {
        const { label, id, param, value, type, filters } = this.props;
        const { checked } = this.state;

        return (() => {
            switch (type) {
                case 'checkbox':
                    return (
                        <Checkbox
                            aria-label={ label }
                            id={ id }
                            isChecked={ checked }
                            label={ label }
                            onChange={ this.handleChange }
                            param={ param }
                            value={ value }
                        />
                    );
                case 'radio':
                    return (
                        <Radio
                            isChecked={ filters[param] === value }
                            aria-label={ label }
                            id={ id }
                            label={ label }
                            name={ param }
                            onChange={ this.handleChange }
                            param={ param }
                            value={ value }
                        />
                    );
            }
        })();
    }
}

FilterInput.propTypes = {
    addRemoveFilters: PropTypes.func,
    className: PropTypes.string,
    currentPage: PropTypes.string,
    id: PropTypes.string.isRequired,
    label: PropTypes.node.isRequired,
    param: PropTypes.string,
    value: PropTypes.any,
    filters: PropTypes.object,
    type: PropTypes.string
};

FilterInput.defaultProps = {
    addRemoveFilters: Function.prototype,
    className: null,
    currentPage: null,
    param: null,
    filters: {},
    type: 'checkbox',
    value: undefined
};

export default FilterInput;
