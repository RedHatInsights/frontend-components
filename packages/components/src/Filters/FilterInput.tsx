import React, { useState, useEffect } from 'react';
import { Checkbox, Radio } from '@patternfly/react-core';

export interface FilterInputProps {
  addRemoveFilters?: (value?: string, param?: string, type?: string, checked?: boolean) => void;
  className?: string;
  currentPage?: string;
  id: string;
  label: React.ReactNode;
  param?: string;
  value?: string;
  filters?: any;
  type?: string;
}

const FilterInput: React.FC<FilterInputProps> = ({ addRemoveFilters, param = '', filters = {}, type = 'checkbox', value, id, label }) => {
  const [checked, setChecked] = useState(false);
  useEffect(() => {
    if (type !== 'radio') {
      if (param in filters && filters?.[param]?.includes(value)) {
        setChecked(true);
      } else {
        setChecked(false);
      }
    }
  }, []);
  const handleChange = () => {
    if (type === 'checkbox') {
      setChecked(!checked);
    }
    addRemoveFilters?.(value, param, type, checked);
  };
  if (type !== 'radio') {
    return (
      <Checkbox aria-label={String(label)} id={id} isChecked={checked} label={label} onChange={handleChange} value={value} ouiaId={String(label)} />
    );
  }
  return (
    <Radio
      isChecked={filters && !!value && !!param && filters[param] === value}
      aria-label={String(label)}
      id={id}
      label={label}
      name={param}
      onChange={handleChange}
      value={value}
      ouiaId={String(label)}
    />
  );
};

export default FilterInput;
