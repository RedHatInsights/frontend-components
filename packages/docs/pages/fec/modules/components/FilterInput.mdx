
# FilterInput

## Props

|name|type|default|description|
|----|----|-------|-----------|
|addRemoveFilters|(value?:string,param?:string,type?:string,checked?:boolean)=&gt;void|||
|className|`string`|||
|currentPage|`string`|||
|id&#42;|`string`|||
|label&#42;|React.ReactNode|||
|param|`string`|`''`||
|value|`string`|||
|filters|`any`|`&#123;&#125;`||
|type|`string`|`'checkbox'`||



