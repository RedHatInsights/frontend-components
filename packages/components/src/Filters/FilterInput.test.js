import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FilterInput from './FilterInput';

describe('FilterInput component', () => {
    it('should render a radio input', () => {
        const wrapper = shallow(
            <FilterInput
                aria-label="label"
                id="id"
                label="label"
                addRemoveFilters={ jest.fn() }
                param="param"
                type="radio"
                value="value"
                filters={ { param: 'value' } }
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render a checkbox input', () => {
        const wrapper = shallow(
            <FilterInput aria-label="label" id="id" label="label" addRemoveFilters={ jest.fn() } param="param" type="checkbox" value="value" />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
