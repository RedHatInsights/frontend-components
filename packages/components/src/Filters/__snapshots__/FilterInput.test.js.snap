// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterInput component should render a checkbox input 1`] = `
<Checkbox
  aria-label="label"
  className=""
  id="id"
  isChecked={false}
  isDisabled={false}
  isValid={true}
  label="label"
  onChange={[Function]}
  param="param"
  value="value"
/>
`;

exports[`FilterInput component should render a radio input 1`] = `
<Radio
  aria-label="label"
  className=""
  id="id"
  isChecked={true}
  isDisabled={false}
  isValid={true}
  label="label"
  name="param"
  onChange={[Function]}
  param="param"
  value="value"
/>
`;
